### PR TITLE
fix(go) update doc to use go mod when installing go-pluginserver

### PR DESCRIPTION
**Pull request by fffonion**
### Review

@reviewer

### Reason
From https://github.com/Kong/docker-kong/issues/458

### CODE REVIEW

**Comment by fffonion** [comment]
This doesn't look right, will rewrite.

**Review by HCloward — APPROVED** [review]
LGTM - thanks @fffonion !

**Comment by fffonion** [comment]
I updated the PR as the context is not accurate before, would you take another review @HCloward ?

File: app/enterprise/2.1.x/go.md [hunk 1]
```diff
@@ -117,6 +117,13 @@ $ go build github.com/Kong/go-pluginserver
 go build -buildmode plugin go-hello.go
 ```
 
+Note: when installing go-pluginserver binary globally user will need to
```
Comment (tlblessing) [inline app/enterprise/2.1.x/go.md:120]:
```suggestion
Note: When installing the `go-pluginserver` binary globally, you  need to
```

File: app/enterprise/2.1.x/go.md [hunk 2]
```diff
@@ -117,6 +117,13 @@ $ go build github.com/Kong/go-pluginserver
...
+enable go.mod support by using:
```
Comment (tlblessing) [inline app/enterprise/2.1.x/go.md:121]:
```suggestion
enable `go.mod` support:
```

**Review by tlblessing — CHANGES_REQUESTED** [review]
Global edit comments

File: app/enterprise/2.2.x/go.md [hunk 3]
```diff
@@ -117,6 +117,13 @@ $ go build github.com/Kong/go-pluginserver
 go build -buildmode plugin go-hello.go
 ```
 
+Note: when installing go-pluginserver binary globally user will need to
```
Comment (tlblessing) [inline app/enterprise/2.2.x/go.md:120]:
```suggestion
Note: When installing the `go-pluginserver` binary globally, you  need to
```

File: app/gateway-oss/2.1.x/go.md [hunk 4]
```diff
@@ -95,6 +95,13 @@ we will assume its name is `go-hello`. You can find example Go plugins [here][go
 go build -buildmode plugin go-hello.go
 ```
 
+Note: when installing go-pluginserver binary globally user will need to
```
Comment (tlblessing) [inline app/gateway-oss/2.1.x/go.md:98]:
```suggestion
Note: When installing the `go-pluginserver` binary globally, you  need to
```

File: app/gateway-oss/2.1.x/go.md [hunk 5]
```diff
@@ -95,6 +95,13 @@ we will assume its name is `go-hello`. You can find example Go plugins [here][go
...
+enable go.mod support by using:
```
Comment (tlblessing) [inline app/gateway-oss/2.1.x/go.md:99]:
```suggestion
enable `go.mod` support:
```

File: app/gateway-oss/2.2.x/go.md [hunk 6]
```diff
@@ -95,6 +95,13 @@ we will assume its name is `go-hello`. You can find example Go plugins [here][go
 go build -buildmode plugin go-hello.go
 ```
 
+Note: when installing go-pluginserver binary globally user will need to
```
Comment (tlblessing) [inline app/gateway-oss/2.2.x/go.md:98]:
```suggestion
Note: When installing the `go-pluginserver` binary globally, you  need to
```

File: app/gateway-oss/2.2.x/go.md [hunk 7]
```diff
@@ -95,6 +95,13 @@ we will assume its name is `go-hello`. You can find example Go plugins [here][go
...
+enable go.mod support by using:
```
Comment (tlblessing) [inline app/gateway-oss/2.2.x/go.md:99]:
```suggestion
enable `go.mod` support:
```

**Review by tlblessing — CHANGES_REQUESTED** [review]
made comments to other versions in this PR, need to check on 2.3.x and 2.4.x now. File is no longer go.md in those versions but is now external. Also this information needs to be ported to the enterprise side. Will add this PR to my pending ticket.

**Comment by fffonion** [comment]
The section on "how to build go-pluginserver" is removed in the external-plugins.md in 2.3/2.4, as now the preferred way to develop and build Go plugins is using the embedded server, which doesn't need the standalone go-pluginserver to built. @tlblessing

**Review by tlblessing — APPROVED** [review]

**Comment by tlblessing** [comment]
There is a broken redirect that was fixed in other versions yesterday. Will fix.

> The section on "how to build go-pluginserver" is removed in the external-plugins.md in 2.3/2.4, as now the preferred way to develop and build Go plugins is using the embedded server, which doesn't need the standalone go-pluginserver to built. @tlblessing

@fffonion I should've remembered that.

**Comment by tlblessing** [comment]
@fffonion can you rebase your PR so it will pick up the redirect fix? Then you can go ahead and merge after a successful build. Thank you for the PR!

**Comment by fffonion** [comment]
@tlblessing  Sure just rebased : )

**Pull request merged**